src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/PostBuildEventValueProviderTests.cs

```diff
@@ -0,0 +1,524 @@
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name=""PostBuild"" AfterTargets=""PostBuildEvent"">
+    <Exec Command=""echo &quot;post build output&quot;"" />
+  </Target>
+
+</Project>
+".AsProjectRootElement();
+            systemUnderTest.SetProperty(string.Empty, root);
+            var stringWriter = new System.IO.StringWriter();
+            root.Save(stringWriter);
+
+            var expected = @"<?xml version=""1.0"" encoding=""utf-16""?>
+<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+</Project>";
+
+            var actual = stringWriter.ToString();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public static void SetPropertyTest_RemoveTarget_WhitespaceCharacter()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name=""PostBuild"" AfterTargets=""PostBuildEvent"">
+    <Exec Command=""echo &quot;post build output&quot;"" />
+  </Target>
+
+</Project>
+".AsProjectRootElement();
+            systemUnderTest.SetProperty("       ", root);
+            var stringWriter = new System.IO.StringWriter();
+            root.Save(stringWriter);
+
+            var expected = @"<?xml version=""1.0"" encoding=""utf-16""?>
+<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+</Project>";
+
+            var actual = stringWriter.ToString();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public static void SetPropertyTest_RemoveTarget_TabCharacter()
+        {
+            var root = @"
+<Project Sdk=""Microsoft.NET.Sdk"">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+
+  <Target Name=""PostBuild"" AfterTargets=""PostBuildEvent"">
+    <Exec Command=""echo &quot;post build output&quot;"" />
+  </Target>
+
+</Project>
+".AsProjectRootElement();
+            systemUnderTest.SetProperty("\t\t\t", root);
+            var stringWriter = new System.IO.StringWriter();
+            root.Save(stringWriter);
+
+            var expected = @"<?xml version=""1.0"" encoding=""utf-16""?>
+<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.1</TargetFramework>
+  </PropertyGroup>
+</Project>";
+
+            var actual = stringWriter.ToString();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public static void SetPropertyTest_RemoveTarget_NewlineCharacter()
```

Comment:
We are not removing the target in this testcase. What is the expected behavior when the property is set to `\r\n`?